src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] arrayList;
+    private int size = 0;
+    private int emptyPlaces = 10;
+    private final double ARRAY_ENLARGEMENT_PERCENTAGE = 0.5;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[emptyPlaces];
+    }
+
+    public void enlarge() {
+        emptyPlaces = (int) (size * ARRAY_ENLARGEMENT_PERCENTAGE);
+        T[] temporaryEnlargedArrayList = (T[]) new Object[size + emptyPlaces];
+        for (int i = 0; i < size; i++) {
+            temporaryEnlargedArrayList[i] = arrayList[i];
+        }
+        arrayList = (T[]) new Object[size + emptyPlaces];
```

Comment:
There is no need to create a new array here since you already created 'temporaryEnlargedArrayList'. You should assign 'arrayList' to 'temporaryEnlargedArrayList' instead of creating a new object.